pkg/token/options.go

```diff
@@ -113,47 +138,75 @@ func (o *Options) UpdateFromEnv() {
 	o.tokenCacheFile = getCacheFileName(o)
 
 	if o.UseAzureRMTerraformEnv {
-		envServicePrincipalClientID = "ARM_CLIENT_ID"
-		envServicePrincipalClientSecret = "ARM_CLIENT_SECRET"
-		envServicePrincipalClientCert = "ARM_CLIENT_CERTIFICATE_PATH"
-		envServicePrincipalClientCertPassword = "ARM_CLIENT_CERTIFICATE_PASSWORD"
-		envTenantID = "ARM_TENANT_ID"
+		if v, ok := os.LookupEnv(terraformClientID); ok {
```

Comment:
suggestion(non-blocking): instead of using a long if-else, maybe we can consider decouple these options into smaller structs, like `AADOptions` / `TerraformOptions`. In those structs, we can make use of env struct libs like this: https://github.com/caarlos0/env to fill in the value. One more benefit is we can decouple the validations into those structs instead of putting in one big `Validate` method.